### PR TITLE
Change Weyl group action to a right action

### PR DESCRIPTION
**Pull request by lgoettgens**
After some discussion with @fingolfin we came to the conclusion, that Weyl groups should not have the special role they have right now, but should follow the Oscar rule that all group actions are from the right, out of consistency.
Other mathematical areas have swallowed the pill as well.

pinging all people that could be affected by this change: @felix-roehrich @janikapeters @gfourier @joschmitt @antonydellavecchia @flenzen

### CODE REVIEW

File: experimental/LieAlgebras/src/WeightLattice.jl [hunk 1]
```diff
@@ -362,7 +360,7 @@ function conjugate_dominant_weight_with_elem!(w::WeightLatticeElem)
 
   # reversing word means it is in short revlex normal form
   # and it is the element taking original w to new w
-  return w, weyl_group(root_system(w))(reverse!(word); normalize=false)
+  return w, weyl_group(root_system(w))(word; normalize=true) # TODO: is normalize=true necessary?
```
Comment (lgoettgens) [inline experimental/LieAlgebras/src/WeightLattice.jl:363]: here

File: experimental/LieAlgebras/src/WeylGroup.jl [hunk 2]
```diff
@@ -719,7 +704,10 @@ function Base.iterate(iter::WeylIteratorNoCopy, state::WeylIteratorNoCopyState)
   if isnothing(state)
     return nothing
   end
-  return state, state
+  # return state, state
+  # TODO: change internals of iterator to return (wt, x) with wt*x == iter.weight instead of the hack below
+  wt, x = state
+  return (wt, inv(x)), state
```
Comment (lgoettgens) [inline experimental/LieAlgebras/src/WeylGroup.jl:710]: and here

**Review by lgoettgens — COMMENTED** [review]
For the two places in the code, where this PR just adds a very stupid (but computationally expensive) workaround (see the two comments below), @felix-roehrich and I just talked and have the following path forward:
1. Merge this PR (it works, and it is better to have the large breaking change in as early as possible)
2. Change the internally used normal form for Weyl group element words (from short revlex to short lex), including basic operations like `*`.
3. Adapt the more complex functionality to the new normal form, get rid of the two workarounds added here.

**Comment by lgoettgens** [comment]
> 2. Change the internally used normal form for Weyl group element words (from short revlex to short lex), including basic operations like `*`.
> 3. Adapt the more complex functionality to the new normal form, get rid of the two workarounds added here.

This was a lot easier to do than I anticipated: it is already finished in https://github.com/oscar-system/Oscar.jl/pull/4375.
But since this PR works on its own, and both have to adapt a non-trivial number of functions, I would like to keep the two PRs separate.

This one here is good to go from my POV. https://github.com/oscar-system/Oscar.jl/pull/4375 will get rebased as soon as this one here is merged.

**Review by fingolfin — APPROVED** [review]

**Comment by fingolfin** [comment]
Fine by me but perhaps let's wait till triage with merging this just in case there is more feedback?

**Comment by lgoettgens** [comment]
Re-running CI to verify that this does not break https://github.com/oscar-system/Oscar.jl/pull/4260 that was merged in the meantime.

**Comment by lgoettgens** [comment]
Triage had no objections. Will merge later this afternoon

**Pull request merged**